src/main/java/formflow/library/data/UserFileRepositoryService.java

```diff
@@ -27,8 +27,8 @@ public UserFileRepositoryService(UserFileRepository repository) {
    * @param userFile the uploadedFile to save, not null
    * @return UUID of the file
    */
-  public UUID save(UserFile userFile) {
-    return repository.save(userFile).getFileId();
+  public UserFile save(UserFile userFile) {
+    return repository.save(userFile);
   }
```

Comment:
Same thing as Submission - it seems better to return a new object, as it may have changed.